base/src/main/java/com/nedap/archie/base/IntervalDurationConverter.java

```diff
@@ -0,0 +1,73 @@
+package com.nedap.archie.base;
+
+
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.time.temporal.TemporalAmount;
+import java.time.temporal.TemporalUnit;
+import java.time.temporal.UnsupportedTemporalTypeException;
+import java.util.Objects;
+
+import static java.time.temporal.ChronoUnit.DAYS;
+
+/**
+ * A tool to convert Durations from TemporalAmounts in the specific use case of Intervals
+ * It handles the unit Month as an exact number of seconds to do a conversion. This is a bit tricky, since
+ * a number of months has a lower number of seconds and an upper number of seconds. but we need some way of doing this...
+ *
+ * This may need a separate method for lower and upper, which returns the lower and upper bounds of the seconds in a month?
+ */
+class IntervalDurationConverter {
```

Comment:
@boeckers  I added this to be able to also compare months/weeks/etc. But it's a bit tricky - the commented out code is a beginning of how you could solve the remaining problems, but I'm not sure we should do this. For now this is already better than it was, because it now works on months and weeks as well.